juju/juju
# [JUJU-4060] Insert controller configuration at bootstrap

This changes the bootstrap variadic options to accept a `database.TxnRunner` instead of a raw `*sql.DB`.

A new `bootstrap` package is added to `domain/externalcontroller` with a function that returns such an argument for inserting controller configuration at bootstrap time.

The agent bootstrap logic includes use of this mechanism to prime the initial configuration. 

Due to the fact that we cannot store complex data types in Dqlite as we can in Mongo, controller configuration articles who's types in the schema were `Tlist` are now `Tstring` under the assumption that they are comma-delimited. This is intended as a temporary measure, pending discussions on more strongly-typed configuration data structures. The change is isolated to a single commit, which can be easily rolled back to restore the list typings.

## QA steps

- Bootstrap to LXD.
- `make repl-install repl`.
- `dqlite -s 127.0.0.1:17666 controller`.
- `select * from controller_config` should indicate we executed the insert.
